### PR TITLE
Add support for Symfony 7

### PR DESCRIPTION
**Pull request by LoicBoursin**
This PR adds support for Symfony 7. The project itself is already ready for SF7+ (cf. https://github.com/jolicode/automapper/pull/18)

I have tested it locally and it works with just this change! (cf. https://symfony.com/blog/new-in-symfony-6-4-build-dir-improvements)

### CODE REVIEW

**Comment by LoicBoursin** [comment]
Hey there @Korbeil ; Can you trigger the CI please? 😁

**Comment by nikophil** [comment]
Done!

Thanks for this PR!

Would you mind adding symfony version to the ci matrix?

**Comment by LoicBoursin** [comment]
@nikophil Done ✅

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -20,6 +20,15 @@ jobs:
...
     name: tests
+    strategy:
+      fail-fast: false
+      matrix:
+        php-versions:
+          - 8.2
+          - 8.3
+        symfony-versions:
+          - ^6.4
+          - ^7.0
```
Comment (nikophil) [inline .github/workflows/ci.yaml:31]:
```suggestion
          - 6.4.*
          - 7.0.*
```

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -31,10 +40,15 @@ jobs:
...
         with:
-          php-version: 8.2
+          php-version: '${{ matrix.php-versions }}'
           coverage: none
           extensions: mbstring, fileinfo, json, intl, dom
       - name: composer install
+        run: composer require \
+          symfony/framework-bundle:${{ matrix.symfony-versions }}  --dev --no-update \
+          symfony/uid:${{ matrix.symfony-versions }} --dev --no-update  \
+          symfony/yaml:${{ matrix.symfony-versions }} --dev --no-update
```
Comment (nikophil) [inline .github/workflows/ci.yaml:50]: hmmm I'm not sure to like how it's done here: if we add another dependency to the bundle, we'll have to think to add it here as well.

maybe we can consider using `symfony/flex` and `SYMFONY_REQUIRE`? just few step to add:
- allow symfony flex plugin in `composer.json`
- add `tools: flex` to `shivammathur/setup-php`
- add the env var `SYMFONY_REQUIRE` to the `composer install` step

**Review by nikophil — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (LoicBoursin) [reply .github/workflows/ci.yaml]: I followed https://symfony.com/doc/current/bundles/best_practices.html#require-a-specific-symfony-version and it looks like it works perfectly !

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (LoicBoursin) [reply .github/workflows/ci.yaml]: Solved in https://github.com/jolicode/automapper-bundle/pull/6/commits/81b539fb00e377c0497ee4124fdc0b40bb29ed86

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (nikophil) [reply .github/workflows/ci.yaml]: it is more or less the same thing that I suggested 👍

**Review by nikophil — APPROVED** [review]

**Comment by Korbeil** [comment]
Thanks for your contribution @LoicBoursin :pray:

**Pull request merged**